packages/core/helper-plugin/src/hooks/useQueryParams.ts

```diff
@@ -3,25 +3,27 @@ import { useCallback, useMemo } from 'react';
 import { parse, stringify } from 'qs';
 import { useHistory, useLocation } from 'react-router-dom';
 
-const useQueryParams = (initialParams) => {
+type JSON = string | number | boolean | null | { [key: string]: JSON } | Array<JSON>;
+
+const useQueryParams = <T extends Record<string, JSON>>(initialParams?: T) => {
   const { search } = useLocation();
   const { push } = useHistory();
 
   const query = useMemo(() => {
     const searchQuery = search.substring(1);
 
     if (!search) {
-      return initialParams;
+      return initialParams || {};
```

Comment:
this could be seen as an API change because we won't return undefined anymore, do you know the effect of this? 🤔